pylint/pyreverse/inspector.py

```diff
@@ -218,7 +218,8 @@ def visit_assignname(self, node):
                     self.visit_module(frame)
 
             current = frame.locals_type[node.name]
-            values = set(node.infer())
+            ann = utils.get_annotation(node)
+            values = {ann} if ann else set(node.infer())
```

Comment:
Here there can be an inference fail, we should catch it (like line 238). Btw it look like the code is similar, maybe we can create a function for it ?